integrations/microstream/health/src/main/java/io/helidon/integrations/microstream/health/MicrostreamHealthCheck.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.microstream.health;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import one.microstream.storage.embedded.types.EmbeddedStorageManager;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Microstream health check.
+ */
+public class MicrostreamHealthCheck implements HealthCheck {

```

Comment:
If you add `META-INF/beans.xml` and annotate this class with `@ApplicationScoped` + create a new constructor with injected values (can be package visible) annotated with `@Inject`, this can work automatically in CDI as well.
The depencencies on required libraries can be in scope `provided` with `optional` set to `true` (and configured as `static` in `module-info.java`)

This can be done as long as the only dependencies missing in SE are annotations.

Reply:
This is problematic as there can be more than one microstream instance. Therefore, I decided against automatic health check creation in this first integration.